### PR TITLE
fixed testAvailabilityQuery34 and testAvailabilityQueryUnavailability28

### PR DESCRIPTION
**Pull request by CippoX**
This is my first fix attempt for this repo. I hope I'm heading in the right direction.

### CODE REVIEW

File: Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift [hunk 1]
```diff
@@ -452,6 +452,18 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
         message: { _ in .joinConditionsUsingComma },
         moveFixIt: { ReplaceTokensFixIt(replaceTokens: $0, replacement: trailingComma) }
       )
+    } else if let unexpected = (node.condition.as(AvailabilityConditionSyntax.self)?.unexpectedBetweenAvailabilitySpecAndRightParen) {
```
Comment (kimdv) [inline Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift:455]: Not sure if you need the left and right paren around 
```suggestion
    } else if let unexpected = node.condition.as(AvailabilityConditionSyntax.self)?.unexpectedBetweenAvailabilitySpecAndRightParen {
```

Comment (CippoX) [reply Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift]: Ty

**Review by ahoppen — COMMENTED** [review]
Thanks for the pull request. This is looking good already. I’d prefer to implement the recovery inside the parser instead of `ParseDiagnosticGenerator.swift` though because that allows us to form a SwiftSyntax tree that still represents the code the user intended to write. 

So, wherever we currently parse the `>=` as unexpected tokens, I would consume the `>=` as unexpected text and then continue parsing  `10.51` as a version.

Could you try that? If something doesn’t make sense to you, let me know.

**Comment by CippoX** [comment]
Hope I understood the request

File: Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift [hunk 2]
```diff
@@ -440,6 +440,23 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: AvailabilityVersionRestrictionSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+    if let unexpected = node.unexpectedBetweenPlatformAndVersion {
```
Comment (ahoppen) [inline Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift:447]: `unexpected` could theoretically be anything if we changed the parser, so just to be sure, I’d suggest to check that `unexpected` really only contains a `>=` operator here:

```swift
unexpected.onlyToken(where: { $0.tokenKind == .binaryOperator(">=") }) != nil
```

**Review by ahoppen — COMMENTED** [review]
One minor comment, otherwise this looks excellent to me. Thank you very much.

**Comment by CippoX** [comment]
I had a little problem with source control 😅 now everything should be ok

**Review by ahoppen — APPROVED** [review]
Great. I’ve got one final request: Could you squash the commits in this PR. It just makes for a nicer Git history.

**Comment by CippoX** [comment]
@ahoppen I'm sorry but looks like something went wrong, can you help me

**Pull request closed without merging**